crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -24,6 +25,11 @@ pub mod graph {
 }
 pub const CORE_3D: &str = graph::NAME;
 
+#[cfg(any(not(feature = "webgl"), not(target_arch = "wasm32")))]
+pub const CORE_3D_DEPTH_FORMAT: TextureFormat = TextureFormat::Depth32FloatStencil8;
```

Comment:
Not sure, but given a comment about support from @Elabajaba above I think we'll make this configurable and someone can use just Depth32Float if they don't want to use deferred. If deferred and forward are used together I'd assume it's better to use Depth32FloatStencil8 rather than blit the depth onto a depth only texture for forward.

Reply:
Yeah, if we're not using the stencil buffer for forward then it'll be cheaper to use Depth32float as Depth32FloatStencil8 uses 64bits.

Reply:
@Elabajaba Do you have a source on it being 64bits? A while back fintelia mentioned:
> ...stencil is never packed together with depth. Just use 32-bit depth + 8-bit stencil

Reply:
For Vulkan it might be up to the implementation? The spec says:

>VK_FORMAT_D32_SFLOAT_S8_UINT specifies a two-component format that has 32 signed float bits in the depth component and 8 unsigned integer bits in the stencil component. There are optionally 24 bits that are unused.

For DX12 they just tell you in the format name (DXGI_FORMAT_D32_FLOAT_S8X24_UINT) that there's 24 padding bits. (though maybe drivers are doing funky stuff?)

For GLES 3 [the docs](https://registry.khronos.org/OpenGL-Refpages/es3.0/html/glTexImage2D.xhtml) say that GL_DEPTH32F_STENCIL8 uses GL_FLOAT_32_UNSIGNED_INT_24_8_REV as its type.

The [Metal docs](https://developer.apple.com/documentation/metal/mtlpixelformat/depth32float_stencil8) say it's 40 bits.

Reply:
Semi-related, [AMD recommends](https://gpuopen.com/learn/rdna-performance-guide/#resources) using a 32bit format for depth instead of a 24bit format as they're the same cost but 32bit has better precision (but I have no idea about combined depth + stencil formats).

Meanwhile Nvidia seems to prefer 24bit depth formats https://developer.nvidia.com/blog/vulkan-dos-donts/

Reply:
I may actually abandon the stencil altogether and use the depth method instead if I can figure it out. I'm having issues with the stencil stuff on DX12. The depth thing would be something like: write material ID to separate attachment of some kind, use that as depth texture in full tri lighting passes and use depth equals for running the frag shader. So it's similar to using the stencil, but then we can just continue to use `Depth32Float` everywhere for depth like we have been.

Reply:
This is the method Nanite is doing, right? Cool!